community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java

```diff
@@ -469,7 +468,19 @@ public static Value decode( long[] valueBlocks, int offset )
 
     public static long[] encodeDateTime( int keyId, long epochSecond, long nanoOfSecond, String zoneId )
     {
-        throw new UnsupportedOperationException( "Cannot yet store DateTime with ZoneID" );
+        int idBits = StandardFormatSettings.PROPERTY_TOKEN_MAXIMUM_ID_BITS;
+        short zoneNumber = TimeZoneMapping.TIME_ZONE_STRING_TO_SHORT.get( zoneId );
+
+        long keyAndType = keyId | (((long) (PropertyType.TEMPORAL.intValue()) << idBits));
+        long temporalTypeBits = TemporalType.TEMPORAL_DATE_TIME.temporalType << (idBits + 4);
+
+        long[] data = new long[3];
```

Comment:
Has this `3` any relation to the one in `calculateNumberOfBlocksUsedForTemporal`? If so, or perhaps even if not, then introducing constants is a good idea and really aids readers of this code.